rust/arrow/src/mod.rs

```diff
@@ -20,6 +20,7 @@ pub mod array_data;
 pub mod bitmap;
 pub mod buffer;
 pub mod builder;
+pub mod compute;
```

Comment:
Chao mentioned that this isn't necessary as we have lib.rs, I've removed it in #3624

Reply:
I actually wasn't sure about this as I remembered @sunchao's previous comment.  We have a `lib.rs` and a `mod.rs` what's the use case for each?  For instance, why is it a good idea to have `csv` in both and not `compute` (not pushing for `compute` to be added but rather wondering what the thinking should be here)?

I tried searching for answers to this but couldn't find anything definitive...

Reply:
lib is required, and I think mod is optional if you prefer declaring top-level modules separately to lib. As we declare `pub mod array` and others in lib, mod becomes redundant.

I think modules that are only declared in lib were an oversight by their author(s).

Reply:
Thanks, it looks like `mod.rs` can be removed.  It's not clear to me that it actually gets compiled because not other files reference it, `lib.rs` references the other modules directly.  `datafusion` and `parquet` do not have a `mod.rs`.